lib/galaxy/webapps/galaxy/api/drs.py

```diff
@@ -0,0 +1,105 @@
+AccessIDParam: str = Path(
+    ..., title="Access ID", description="The access ID of the access method for objects, unused in Galaxy."
+)
+
+DRS_SERVICE_NAME = "Galaxy DRS API"
+DRS_SERVICE_DESCRIPTION = "Serves Galaxy datasets according to the GA4GH DRS specification"
+
+
+@router.cbv
+class DrsApi:
+    service: DatasetsService = depends(DatasetsService)
+    app: StructuredApp = depends(StructuredApp)
+
+    @router.get("/ga4gh/drs/v1/service-info")
+    def service_info(self, request: Request) -> Service:
+        components = request.url.components
+        hostname = components.hostname
+        assert hostname
+        default_organization_id = ".".join(reversed(hostname.split(".")))
+        config = self.app.config
+        organization_id = config.ga4gh_service_id or default_organization_id
+        organization_name = config.ga4gh_service_organization_name or organization_id
+        organization_url = config.ga4gh_service_organization_url or f"{components.scheme}://{components.netloc}"
+
+        organization = ServiceOrganization(
+            url=organization_url,
+            name=organization_name,
+        )
+        service_type = ServiceType(
+            group="org.ga4gh",
+            artifact="drs",
+            version="1.2.0",
+        )
+        return Service(
+            id=organization_id + ".drs",
+            name=DRS_SERVICE_NAME,
+            description=DRS_SERVICE_DESCRIPTION,
+            organization=organization,
+            type=service_type,
+            version=VERSION,
+        )
+
+    @router.get("/ga4gh/drs/v1/objects/{object_id}")
+    @router.post("/ga4gh/drs/v1/objects/{object_id}")  # spec specifies both get and post should work.
+    def get_object(
+        self,
+        request: Request,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        object_id: EncodedDatabaseIdField = ObjectIDParam,
+    ) -> DrsObject:
+        return self.service.get_drs_object(trans, object_id, request_url=request.url)
+
+    @router.get("/ga4gh/drs/v1/objects/{object_id}/access/{access_id}")
+    @router.post("/ga4gh/drs/v1/objects/{object_id}/access/{access_id}")
+    def get_access_url(
+        self,
+        request: Request,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        object_id: EncodedDatabaseIdField = ObjectIDParam,
+        access_id: str = AccessIDParam,
+    ):
+        raise ObjectNotFound("Access IDs are not implemented for this DRS server")
+
+    @router.get(
+        "/api/drs_download/{object_id}",
+        response_class=FileResponse,
+    )
+    def download(
+        self, trans: ProvidesHistoryContext = DependsOnTrans, object_id: EncodedDatabaseIdField = ObjectIDParam
+    ):
+        display_data, headers = self.service.display(trans, object_id, False, None, None, True)
+        assert isinstance(display_data, IOBase)
+        return FileResponse(file_name, headers=headers)
```

Comment:
I'm sorry for the bad suggestion, should have been:
```suggestion
        return FileResponse(getattr(display_data, "name"), headers=headers)
```
The getattr prevents mypy from complaining, there should always be a `.name` attribute.